lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -111,6 +114,8 @@ struct FIRRTLTypesLowering : public LowerFIRRTLTypesBase<FIRRTLTypesLowering>,
   Value getBundleLowering(Value oldValue, StringRef flatField);
   void getAllBundleLowerings(Value oldValue, SmallVectorImpl<Value> &results);
 
+  StringRef getArgAttrName(unsigned newArgNumber);
```

Comment:
This can be more self contained and just as efficient if you make this return an MLIR identifier:

```
Identifier FIRRTLTypesLowering::getArgAttrName(unsigned newArgNumber) {
  SmallString<16> argNameBuffer;
  mlir::impl::getArgAttrName(newArgNumber, argNameBuffer);
  return Identifier::get(argNameBuffer, context);
}
```

That will eliminate the argNameBuffer.